File: Front/WebShop-Frontend/src/app/models/sign-up-request.ts
export class SignUpRequest {
    firstName: string;
    lastName: string;
    username: string;
    password: string;
    email: string;
    phoneNumber: string;
    city: string;
    avatarUri:string;

    constructor(firstName:string, lastName:string, username:string, password:string, email:string, phoneNumber:string, city:string, avatarUri:string){
        this.firstName = firstName;
        this.lastName = lastName;
        this.username = username;
        this.password = password;
        this.email = email;
        this.phoneNumber = phoneNumber;
        this.city = city;
        this.avatarUri = avatarUri;
    }
}

File: Back/WebShop-Backend/src/main/java/com/example/WebShop/models/requests/SignUpRequest.java
package com.example.WebShop.models.requests;

import jakarta.persistence.Basic;
import jakarta.persistence.Column;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.Data;
import org.springframework.lang.Nullable;

@Data
public class SignUpRequest {
    @NotBlank
    private String firstName;
    @NotBlank
    private String lastName;
    @NotBlank
    private String username;
    @NotBlank
    private String password;
    @NotBlank
    @Email
    private String email;
    @NotBlank
    private String phoneNumber;
    @NotBlank
    private String city;
    @NotBlank
    private String avatarUri;
}